File: Cap xB/Calculadora V4/calculadoraV4.pl
use strict;
use warnings;

sub ENTRADA {
    my ($numero) = @_;
    print "Entre o ${numero}o. valor: ";
    my $input = <STDIN>;
    chomp($input);
    return $input;
}

sub SAIDA {
    my ($resultado) = @_;
    print "\n";
    print "O resultado da operacao equivale a: " . sprintf("%.2f", $resultado) . ".\n";
    print "\n";
}

sub CALCULO {
    my ($X, $Y, $OPERADOR) = @_;
    if ($OPERADOR eq '+') {
        return $X + $Y;
    } elsif ($OPERADOR eq '-') {
        return $X - $Y;
    } elsif ($OPERADOR eq '*') {
        return $X * $Y;
    } elsif ($OPERADOR eq '/') {
        return $X / $Y;
    }
}

sub ROTCALC {
    my ($OPERACAO) = @_;
    if ($OPERACAO eq '+') {
        print "\nRotina de Adicao\n\n";
    } elsif ($OPERACAO eq '-') {
        print "\nRotina de Subtracao\n\n";
    } elsif ($OPERACAO eq '*') {
        print "\nRotina de Multiplicacao\n\n";
    } elsif ($OPERACAO eq '/') {
        print "\nRotina de Divisao\n\n";
    }
    my $A = ENTRADA('1');
    my $B = ENTRADA('2');
    if ($OPERACAO eq '/') {
        if ($B == 0) {
            print "\n";
            print "O resultado da operacao equivale a: ERRO.\n";
            print "\n";
        } else {
            my $R = CALCULO($A, $B, '/');
            SAIDA($R);
        }
    } else {
        my $R = CALCULO($A, $B, $OPERACAO);
        SAIDA($R);
    }
}

my $OPCAO = 0;
while ($OPCAO != 5) {
    print "CALCULADORA  - V4\n";
    print "=================\n";
    print "\n";
    print "1 - Adicao\n";
    print "2 - Subtracao\n";
    print "3 - Multiplicacao\n";
    print "4 - Divisao\n";
    print "5 - Fim de Programa\n";
    print "\n";
    print "Escolha uma opcao: ";
    $OPCAO = <STDIN>;
    chomp($OPCAO);
    if ($OPCAO != 5) {
        if ($OPCAO == 1) {
            ROTCALC('+');
        } elsif ($OPCAO == 2) {
            ROTCALC('-');
        } elsif ($OPCAO == 3) {
            ROTCALC('*');
        } elsif ($OPCAO == 4) {
            ROTCALC('/');
        } else {
            print "\n";
            print "Opcao invalida - Tente novamente.\n";
            print "\n";
        }
    }
}

File: Cap xB/calculadoraV4.scala
import scala.io.StdIn.readLine
import scala.math.BigDecimal.RoundingMode
import java.text.DecimalFormat

object CALCULADORAV4 {
  var R, A, B: Double = 0.0
  var OPCAO: Int = 0

  def ENTRADA(): Unit = {
    print("Entre o 1o. valor: ")
    A = readLine().toDouble
    print("Entre o 2o. valor: ")
    B = readLine().toDouble
  }

  def SAIDA(): Unit = {
    val decimalFormat = new DecimalFormat("#0.00")
    println(s"\nO resultado da operacao equivale a: ${decimalFormat.format(R)}.\n")
  }

  def CALCULO(X: Double, Y: Double, OPERADOR: Char): Double = {
    OPERADOR match {
      case '+' => X + Y
      case '-' => X - Y
      case '*' => X * Y
      case '/' => X / Y
      case _ => 0.0
    }
  }

  def ROTCALC(OPERAÇÃO: Char): Unit = {
    OPERAÇÃO match {
      case '+' => println("\nRotina de Adicao\n")
      case '-' => println("\nRotina de Subtracao\n")
      case '*' => println("\nRotina de Multiplicacao\n")
      case '/' => println("\nRotina de Divisao\n")
    }
    ENTRADA()
    if (OPERAÇÃO == '/') {
      if (B == 0) {
        println("\nO resultado da operacao equivale a: ERRO.\n")
      } else {
        R = CALCULO(A, B, '/')
        SAIDA()
      }
    }
    if (OPERAÇÃO != '/') {
      R = CALCULO(A, B, OPERAÇÃO)
      SAIDA()
    }
  }

  def main(args: Array[String]): Unit = {
    while (OPCAO != 5) {
      println("CALCULADOEA - V4")
      println("================")
      println()
      println("1 – Adicao")
      println("2 – Subtracao")
      println("3 – Multiplicacao")
      println("4 – Divisao")
      println("5 – Fim de Programa")
      println()
      print("Escolha uma opcao: ")
      OPCAO = readLine().toInt
      if (OPCAO != 5) {
        OPCAO match {
          case 1 => ROTCALC('+')
          case 2 => ROTCALC('-')
          case 3 => ROTCALC('*')
          case 4 => ROTCALC('/')
          case _ => println("\nOpcao invalida - Tente novamente.\n")
        }
      }
    }
  }
}
